pkg/splunk/enterprise/util.go

```diff
@@ -2272,3 +2272,50 @@ func getApplicablePodNameForK8Probes(cr splcommon.MetaObject, ordinalIdx int32)
 	}
 	return fmt.Sprintf("splunk-%s-%s-%d", cr.GetName(), podType, ordinalIdx)
 }
+
+// getClusterManagerCurrentImage gets the image of the pods of the clusterManager before any upgrade takes place,
+// returns the image, and error if something goes wrong
+func getCurrentImage(ctx context.Context, c splcommon.ControllerClient, cr splcommon.MetaObject, instanceType InstanceType) (string, error) {
+
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      GetSplunkStatefulsetName(instanceType, cr.GetName()),
+	}
+	statefulSet := &appsv1.StatefulSet{}
+	err := c.Get(ctx, namespacedName, statefulSet)
+	if err != nil {
+		return "", err
+	}
+
+	image := statefulSet.Spec.Template.Spec.Containers[0].Image
+
+	return image, nil
+
+}
+
+// changeAnnotations updates the checkUpdateImage field of the CLuster Manager Annotations to trigger the reconcile loop
+// on update, and returns error if something is wrong
+func changeAnnotations(ctx context.Context, c splcommon.ControllerClient, image string, cr splcommon.MetaObject) error {
+
+	annotations := cr.GetAnnotations()
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+	if _, ok := annotations["splunk/image-tag"]; ok {
```

Comment:
The two return values of the annotations map lookup are value, bool(whether there is an entry in the map). The second lookup is unnecessary? We can have something like:

```
if val, ok := annotations["splunk/image-tag"]; ok && (val == img) {
    return nil
}
```

Reply:
What I meant here is that we can avoid the nested if and a second evaluation of `annotations["splunk/image-tag"]`. I would keep it as stated above to keep the code cleaner and still achieve the same functionality.